### ISSUE TITLE
Container cannot start if etcd does not listen on 127.0.0.1 address

### ISSUE THREAD

**Issue by sitya** [deis/deis#3280]
As it is advised, on our hosts etcd listens on only the private network addresses (10.x.x.x:4001), but in this case it is not able to start any container, because in the generated ServiceUnit files there are no `-C $ETCD` option for the `etcdctl` commands, so the image cannot be found.

Like:
```
[Unit]
Description=lambda-unicycle_v2.cmd.1

[Service]
ExecStartPre=/bin/sh -c "IMAGE=$(etcdctl get /deis/registry/host 2>&1):$(etcdctl get /deis/registry/port 2>&1)/lambda-unicycle:v2; docker pull $IMAGE"
ExecStartPre=/bin/sh -c "docker inspect lambda-unicycle_v2.cmd.1 >/dev/null 2>&1 && docker rm -f lambda-unicycle_v2.cmd.1 || true"
ExecStart=/bin/sh -c "IMAGE=$(etcdctl get /deis/registry/host 2>&1):$(etcdctl get /deis/registry/port 2>&1)/lambda-unicycle:v2; port=$(docker inspect -f '{{range $k, $v := .ContainerConfig.ExposedPorts }}{{$k}}{{end}}' $IMAGE | cut -d/ -f1) ; docker run --name lambda-unicycle_v2.cmd.1    -P -e PORT=$port $IMAGE "
ExecStop=/usr/bin/docker stop lambda-unicycle_v2.cmd.1
ExecStop=/usr/bin/docker rm -f lambda-unicycle_v2.cmd.1
TimeoutStartSec=20m
TimeoutStopSec=10
RestartSec=5
Restart=on-failure
```

I guess it is generated via the https://github.com/deis/deis/blob/master/controller/scheduler/fleet.py file.

**Comment by carmstrong** [comment]
@sitya I'm not sure I understand the issue. Our cloud-config uses `$private_ipv4` (which is set by CoreOS) to bind to: https://github.com/deis/deis/blob/master/contrib/coreos/user-data.example#L8-L9

Can you investigate `/etc/environment`? Perhaps your `PUBLIC_IPV4` and `PRIVATE_IPV4` are not being set properly?

**Comment by sitya** [comment]
Hi,
the `addr` and `peer-addr` work fine, but without setting `bind-addr` and `peer-bind-addr` explicitly to `PRIVATE_IPV4` etcd will answer properly on the `http://$PUBLIC_IPV4:4001` as well. And it is problem. But if I set these options explicitly to `PRIVATE_IPV4` it won't be listening on 127.0.0.1. :-(

I think,  it will be solved with [upgrading to etcd 2.0](https://github.com/coreos/etcd/blob/master/Documentation/clustering.md#04-to-20-migration-guide), with the `listening-*` options.

Till CoreOS and Deis will use etcd 2.0.x, I configured [Deis firewall](http://docs.deis.io/en/latest/managing_deis/security_considerations/) to prevent connect to etcd using the `PUBLIC_IPV4` address, but I am pretty sure, that the sustainable soulution will be the using of client certification to connect etcd by Deis, as we talked about in #3201 issue.
